src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,49 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.model.Workable;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private MachineProducer bulldozerProducer = new BulldozerProducer();
```

Comment:
The `bulldozerProducer` should be parameterized with `Bulldozer` to ensure type safety and proper functionality. Change `MachineProducer` to `MachineProducer<Bulldozer>`.